docs/docs/creating-dynamically-rendered-navigation.md

```diff
@@ -2,7 +2,175 @@
 title: Creating Dynamically-Rendered Navigation
 ---
 
-This is a stub. Help our community expand it.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Creating dynamic navigation in Gatsby
+
+At times you will want to be able to edit your website's navigation in response to a *change in requirements*. To achieve this, you can use Gatsby to dynamically generate your navigation. Where you store the data for your navigation can be anywhere - a backend API, CMS, headless CMS or even the filesystem.
+
+What this doc will cover:
+
+ - Adding data to your siteMetadata
+ - Querying it using GraphQL
+ - Pulling the data into a component using StaticQuery
+ - Dynamically generating a navigation component
+
+You will be using `gatsby-config.js` to store the data for your links. `gatsby-config.js` is a file used for configuring gatsby located in the root path of every Gatsby project. A plain old javascript object is exported from this file, this object contains the `siteMetadata` object which you can query through graphql when generating your static pages. 
+
+This guide will use the Gatsby starter project `gatsby-starter-default`, which can be downloaded through the Gatsby command line interface tool using the command `gatsby new [project-name] https://github.com/gatsbyjs/gatsby-starter-default#v2`.  
+
+### Creating the link data
+First, locate the `gatsby-config.js` file in the root directory of your project. Inside the `siteMetadata` object, add an array of menu link objects. These objects should contain two properties: name and link. Name is the name of your navigation item, link is the page which will be navigated to when a menu item is clicked.
```

Comment:
I think it makes sense to format properties with back ticks, like `name` and `link`.